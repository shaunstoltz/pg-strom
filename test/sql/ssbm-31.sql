SET pg_strom.regression_test_mode = on;
SET search_path = pgstrom_regress,public;
SET pg_strom.debug_kernel_source = off;
--Q3_1
SET pg_strom.enabled = on;
select c_nation, s_nation, d_year, sum(lo_revenue)
as revenue from customer, lineorder, supplier, date1
where lo_custkey = c_custkey
and lo_suppkey = s_suppkey
and lo_orderdate = d_datekey
and c_region = 'ASIA'  and s_region = 'ASIA'
and d_year >= 1992 and d_year <= 1997
  group by c_nation, s_nation, d_year
  order by d_year asc, revenue desc;

SET pg_strom.enabled = off;
select c_nation, s_nation, d_year, sum(lo_revenue)
as revenue from customer, lineorder, supplier, date1
where lo_custkey = c_custkey
and lo_suppkey = s_suppkey
and lo_orderdate = d_datekey
and c_region = 'ASIA'  and s_region = 'ASIA'
and d_year >= 1992 and d_year <= 1997
  group by c_nation, s_nation, d_year
  order by d_year asc, revenue desc;
