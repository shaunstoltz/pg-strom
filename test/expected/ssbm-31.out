SET pg_strom.regression_test_mode = on;
SET search_path = pgstrom_regress,public;
SET pg_strom.debug_kernel_source = off;
--Q3_1
SET pg_strom.enabled = on;
select c_nation, s_nation, d_year, sum(lo_revenue)
as revenue from customer, lineorder, supplier, date1
where lo_custkey = c_custkey
and lo_suppkey = s_suppkey
and lo_orderdate = d_datekey
and c_region = 'ASIA'  and s_region = 'ASIA'
and d_year >= 1992 and d_year <= 1997
  group by c_nation, s_nation, d_year
  order by d_year asc, revenue desc;
    c_nation     |    s_nation     | d_year |   revenue   
-----------------+-----------------+--------+-------------
 VIETNAM         | INDONESIA       |   1992 | 11366763253
 CHINA           | CHINA           |   1992 | 11331799504
 INDONESIA       | INDONESIA       |   1992 | 11273043073
 CHINA           | INDONESIA       |   1992 | 11194488299
 CHINA           | INDIA           |   1992 | 11023884677
 JAPAN           | INDONESIA       |   1992 | 10937074801
 JAPAN           | INDIA           |   1992 | 10902032622
 INDIA           | INDIA           |   1992 | 10862910313
 INDIA           | INDONESIA       |   1992 | 10850644940
 VIETNAM         | INDIA           |   1992 | 10747853953
 CHINA           | VIETNAM         |   1992 | 10638323504
 INDONESIA       | INDIA           |   1992 | 10617167608
 CHINA           | JAPAN           |   1992 | 10607873798
 INDIA           | JAPAN           |   1992 | 10597082573
 JAPAN           | CHINA           |   1992 | 10468158460
 INDIA           | CHINA           |   1992 | 10447770903
 VIETNAM         | CHINA           |   1992 | 10361913116
 INDONESIA       | CHINA           |   1992 | 10357643549
 JAPAN           | JAPAN           |   1992 | 10223274694
 VIETNAM         | JAPAN           |   1992 | 10220736691
 INDONESIA       | JAPAN           |   1992 | 10156661557
 JAPAN           | VIETNAM         |   1992 |  9949085926
 INDONESIA       | VIETNAM         |   1992 |  9864722711
 INDIA           | VIETNAM         |   1992 |  9791159994
 VIETNAM         | VIETNAM         |   1992 |  9509842379
 JAPAN           | INDONESIA       |   1993 | 11651672431
 INDONESIA       | INDONESIA       |   1993 | 11332614684
 CHINA           | INDONESIA       |   1993 | 11309369050
 VIETNAM         | INDONESIA       |   1993 | 11294899293
 INDIA           | INDONESIA       |   1993 | 11090769190
 VIETNAM         | INDIA           |   1993 | 11071131347
 CHINA           | JAPAN           |   1993 | 10945264180
 INDONESIA       | INDIA           |   1993 | 10944412426
 INDIA           | INDIA           |   1993 | 10825780170
 INDONESIA       | JAPAN           |   1993 | 10798279339
 VIETNAM         | CHINA           |   1993 | 10730594565
 INDONESIA       | CHINA           |   1993 | 10687852071
 CHINA           | INDIA           |   1993 | 10576402660
 JAPAN           | INDIA           |   1993 | 10536550641
 JAPAN           | CHINA           |   1993 | 10513550423
 INDIA           | JAPAN           |   1993 | 10411559083
 CHINA           | CHINA           |   1993 | 10392210671
 INDIA           | CHINA           |   1993 | 10370834532
 VIETNAM         | JAPAN           |   1993 | 10368753427
 JAPAN           | JAPAN           |   1993 | 10343303635
 CHINA           | VIETNAM         |   1993 | 10303253170
 VIETNAM         | VIETNAM         |   1993 | 10103334047
 INDIA           | VIETNAM         |   1993 |  9976310237
 JAPAN           | VIETNAM         |   1993 |  9900951879
 INDONESIA       | VIETNAM         |   1993 |  9892818028
 CHINA           | INDONESIA       |   1994 | 11547773839
 CHINA           | INDIA           |   1994 | 11286983804
 INDONESIA       | INDONESIA       |   1994 | 11264564092
 INDIA           | INDONESIA       |   1994 | 11151776423
 JAPAN           | INDONESIA       |   1994 | 11087068635
 CHINA           | CHINA           |   1994 | 10996905155
 INDONESIA       | INDIA           |   1994 | 10944944139
 INDIA           | INDIA           |   1994 | 10938802313
 VIETNAM         | INDONESIA       |   1994 | 10932906002
 JAPAN           | INDIA           |   1994 | 10822644618
 INDIA           | JAPAN           |   1994 | 10790834037
 INDONESIA       | JAPAN           |   1994 | 10779316061
 VIETNAM         | CHINA           |   1994 | 10599927528
 INDONESIA       | CHINA           |   1994 | 10544190577
 JAPAN           | JAPAN           |   1994 | 10521774751
 CHINA           | VIETNAM         |   1994 | 10473101788
 CHINA           | JAPAN           |   1994 | 10462091639
 INDONESIA       | VIETNAM         |   1994 | 10375182910
 VIETNAM         | JAPAN           |   1994 | 10355550499
 JAPAN           | CHINA           |   1994 | 10349345683
 INDIA           | CHINA           |   1994 | 10346949537
 INDIA           | VIETNAM         |   1994 | 10339565008
 JAPAN           | VIETNAM         |   1994 | 10318022285
 VIETNAM         | INDIA           |   1994 | 10053337368
 VIETNAM         | VIETNAM         |   1994 |  9946315606
 INDIA           | INDONESIA       |   1995 | 11446599922
 CHINA           | INDONESIA       |   1995 | 11401808364
 CHINA           | JAPAN           |   1995 | 11370170375
 CHINA           | INDIA           |   1995 | 11157260361
 INDONESIA       | INDONESIA       |   1995 | 11121798405
 VIETNAM         | INDONESIA       |   1995 | 11090077131
 INDONESIA       | INDIA           |   1995 | 10928540142
 VIETNAM         | JAPAN           |   1995 | 10806539206
 JAPAN           | INDONESIA       |   1995 | 10794468592
 JAPAN           | INDIA           |   1995 | 10781463509
 VIETNAM         | INDIA           |   1995 | 10772443445
 INDIA           | INDIA           |   1995 | 10766284475
 INDONESIA       | JAPAN           |   1995 | 10622882310
 INDIA           | CHINA           |   1995 | 10577984844
 CHINA           | CHINA           |   1995 | 10534105501
 INDONESIA       | CHINA           |   1995 | 10524605629
 INDONESIA       | VIETNAM         |   1995 | 10422705118
 JAPAN           | JAPAN           |   1995 | 10420733797
 JAPAN           | CHINA           |   1995 | 10414807621
 VIETNAM         | CHINA           |   1995 | 10362785447
 CHINA           | VIETNAM         |   1995 | 10342184656
 INDIA           | JAPAN           |   1995 | 10338126072
 INDIA           | VIETNAM         |   1995 | 10320472377
 VIETNAM         | VIETNAM         |   1995 | 10291677053
 JAPAN           | VIETNAM         |   1995 | 10259020417
 CHINA           | INDONESIA       |   1996 | 11718329138
 VIETNAM         | INDONESIA       |   1996 | 11256192187
 JAPAN           | INDONESIA       |   1996 | 11196557220
 INDONESIA       | JAPAN           |   1996 | 11107350490
 CHINA           | JAPAN           |   1996 | 11034434949
 INDONESIA       | INDONESIA       |   1996 | 11031182183
 INDIA           | INDONESIA       |   1996 | 11009006499
 VIETNAM         | INDIA           |   1996 | 10947513795
 VIETNAM         | JAPAN           |   1996 | 10927373056
 INDONESIA       | CHINA           |   1996 | 10832876231
 INDIA           | INDIA           |   1996 | 10803769909
 INDIA           | CHINA           |   1996 | 10799869583
 INDONESIA       | INDIA           |   1996 | 10756005575
 CHINA           | INDIA           |   1996 | 10658108154
 VIETNAM         | CHINA           |   1996 | 10609181278
 CHINA           | VIETNAM         |   1996 | 10431194088
 CHINA           | CHINA           |   1996 | 10397959049
 INDIA           | JAPAN           |   1996 | 10367024603
 JAPAN           | CHINA           |   1996 | 10360789892
 VIETNAM         | VIETNAM         |   1996 | 10276179131
 INDIA           | VIETNAM         |   1996 | 10250670880
 JAPAN           | JAPAN           |   1996 | 10238658407
 JAPAN           | INDIA           |   1996 | 10207767222
 JAPAN           | VIETNAM         |   1996 | 10035187510
 INDONESIA       | VIETNAM         |   1996 |  9914928892
 JAPAN           | INDONESIA       |   1997 | 11482072101
 CHINA           | INDONESIA       |   1997 | 11432002557
 INDIA           | INDONESIA       |   1997 | 11418742656
 INDONESIA       | INDONESIA       |   1997 | 11266595254
 VIETNAM         | INDONESIA       |   1997 | 11107866642
 INDONESIA       | INDIA           |   1997 | 10901363844
 INDIA           | CHINA           |   1997 | 10857283915
 CHINA           | JAPAN           |   1997 | 10793325412
 JAPAN           | INDIA           |   1997 | 10773028622
 CHINA           | CHINA           |   1997 | 10772225760
 INDONESIA       | CHINA           |   1997 | 10699078012
 INDONESIA       | JAPAN           |   1997 | 10697685336
 VIETNAM         | INDIA           |   1997 | 10659789598
 INDIA           | INDIA           |   1997 | 10594944414
 CHINA           | INDIA           |   1997 | 10558080684
 INDONESIA       | VIETNAM         |   1997 | 10538588657
 VIETNAM         | JAPAN           |   1997 | 10508662011
 CHINA           | VIETNAM         |   1997 | 10419891376
 INDIA           | JAPAN           |   1997 | 10417917258
 JAPAN           | CHINA           |   1997 | 10380170603
 VIETNAM         | CHINA           |   1997 | 10347399889
 JAPAN           | JAPAN           |   1997 | 10118734544
 JAPAN           | VIETNAM         |   1997 |  9925115976
 VIETNAM         | VIETNAM         |   1997 |  9874907444
 INDIA           | VIETNAM         |   1997 |  9862710061
(150 rows)

SET pg_strom.enabled = off;
select c_nation, s_nation, d_year, sum(lo_revenue)
as revenue from customer, lineorder, supplier, date1
where lo_custkey = c_custkey
and lo_suppkey = s_suppkey
and lo_orderdate = d_datekey
and c_region = 'ASIA'  and s_region = 'ASIA'
and d_year >= 1992 and d_year <= 1997
  group by c_nation, s_nation, d_year
  order by d_year asc, revenue desc;
    c_nation     |    s_nation     | d_year |   revenue   
-----------------+-----------------+--------+-------------
 VIETNAM         | INDONESIA       |   1992 | 11366763253
 CHINA           | CHINA           |   1992 | 11331799504
 INDONESIA       | INDONESIA       |   1992 | 11273043073
 CHINA           | INDONESIA       |   1992 | 11194488299
 CHINA           | INDIA           |   1992 | 11023884677
 JAPAN           | INDONESIA       |   1992 | 10937074801
 JAPAN           | INDIA           |   1992 | 10902032622
 INDIA           | INDIA           |   1992 | 10862910313
 INDIA           | INDONESIA       |   1992 | 10850644940
 VIETNAM         | INDIA           |   1992 | 10747853953
 CHINA           | VIETNAM         |   1992 | 10638323504
 INDONESIA       | INDIA           |   1992 | 10617167608
 CHINA           | JAPAN           |   1992 | 10607873798
 INDIA           | JAPAN           |   1992 | 10597082573
 JAPAN           | CHINA           |   1992 | 10468158460
 INDIA           | CHINA           |   1992 | 10447770903
 VIETNAM         | CHINA           |   1992 | 10361913116
 INDONESIA       | CHINA           |   1992 | 10357643549
 JAPAN           | JAPAN           |   1992 | 10223274694
 VIETNAM         | JAPAN           |   1992 | 10220736691
 INDONESIA       | JAPAN           |   1992 | 10156661557
 JAPAN           | VIETNAM         |   1992 |  9949085926
 INDONESIA       | VIETNAM         |   1992 |  9864722711
 INDIA           | VIETNAM         |   1992 |  9791159994
 VIETNAM         | VIETNAM         |   1992 |  9509842379
 JAPAN           | INDONESIA       |   1993 | 11651672431
 INDONESIA       | INDONESIA       |   1993 | 11332614684
 CHINA           | INDONESIA       |   1993 | 11309369050
 VIETNAM         | INDONESIA       |   1993 | 11294899293
 INDIA           | INDONESIA       |   1993 | 11090769190
 VIETNAM         | INDIA           |   1993 | 11071131347
 CHINA           | JAPAN           |   1993 | 10945264180
 INDONESIA       | INDIA           |   1993 | 10944412426
 INDIA           | INDIA           |   1993 | 10825780170
 INDONESIA       | JAPAN           |   1993 | 10798279339
 VIETNAM         | CHINA           |   1993 | 10730594565
 INDONESIA       | CHINA           |   1993 | 10687852071
 CHINA           | INDIA           |   1993 | 10576402660
 JAPAN           | INDIA           |   1993 | 10536550641
 JAPAN           | CHINA           |   1993 | 10513550423
 INDIA           | JAPAN           |   1993 | 10411559083
 CHINA           | CHINA           |   1993 | 10392210671
 INDIA           | CHINA           |   1993 | 10370834532
 VIETNAM         | JAPAN           |   1993 | 10368753427
 JAPAN           | JAPAN           |   1993 | 10343303635
 CHINA           | VIETNAM         |   1993 | 10303253170
 VIETNAM         | VIETNAM         |   1993 | 10103334047
 INDIA           | VIETNAM         |   1993 |  9976310237
 JAPAN           | VIETNAM         |   1993 |  9900951879
 INDONESIA       | VIETNAM         |   1993 |  9892818028
 CHINA           | INDONESIA       |   1994 | 11547773839
 CHINA           | INDIA           |   1994 | 11286983804
 INDONESIA       | INDONESIA       |   1994 | 11264564092
 INDIA           | INDONESIA       |   1994 | 11151776423
 JAPAN           | INDONESIA       |   1994 | 11087068635
 CHINA           | CHINA           |   1994 | 10996905155
 INDONESIA       | INDIA           |   1994 | 10944944139
 INDIA           | INDIA           |   1994 | 10938802313
 VIETNAM         | INDONESIA       |   1994 | 10932906002
 JAPAN           | INDIA           |   1994 | 10822644618
 INDIA           | JAPAN           |   1994 | 10790834037
 INDONESIA       | JAPAN           |   1994 | 10779316061
 VIETNAM         | CHINA           |   1994 | 10599927528
 INDONESIA       | CHINA           |   1994 | 10544190577
 JAPAN           | JAPAN           |   1994 | 10521774751
 CHINA           | VIETNAM         |   1994 | 10473101788
 CHINA           | JAPAN           |   1994 | 10462091639
 INDONESIA       | VIETNAM         |   1994 | 10375182910
 VIETNAM         | JAPAN           |   1994 | 10355550499
 JAPAN           | CHINA           |   1994 | 10349345683
 INDIA           | CHINA           |   1994 | 10346949537
 INDIA           | VIETNAM         |   1994 | 10339565008
 JAPAN           | VIETNAM         |   1994 | 10318022285
 VIETNAM         | INDIA           |   1994 | 10053337368
 VIETNAM         | VIETNAM         |   1994 |  9946315606
 INDIA           | INDONESIA       |   1995 | 11446599922
 CHINA           | INDONESIA       |   1995 | 11401808364
 CHINA           | JAPAN           |   1995 | 11370170375
 CHINA           | INDIA           |   1995 | 11157260361
 INDONESIA       | INDONESIA       |   1995 | 11121798405
 VIETNAM         | INDONESIA       |   1995 | 11090077131
 INDONESIA       | INDIA           |   1995 | 10928540142
 VIETNAM         | JAPAN           |   1995 | 10806539206
 JAPAN           | INDONESIA       |   1995 | 10794468592
 JAPAN           | INDIA           |   1995 | 10781463509
 VIETNAM         | INDIA           |   1995 | 10772443445
 INDIA           | INDIA           |   1995 | 10766284475
 INDONESIA       | JAPAN           |   1995 | 10622882310
 INDIA           | CHINA           |   1995 | 10577984844
 CHINA           | CHINA           |   1995 | 10534105501
 INDONESIA       | CHINA           |   1995 | 10524605629
 INDONESIA       | VIETNAM         |   1995 | 10422705118
 JAPAN           | JAPAN           |   1995 | 10420733797
 JAPAN           | CHINA           |   1995 | 10414807621
 VIETNAM         | CHINA           |   1995 | 10362785447
 CHINA           | VIETNAM         |   1995 | 10342184656
 INDIA           | JAPAN           |   1995 | 10338126072
 INDIA           | VIETNAM         |   1995 | 10320472377
 VIETNAM         | VIETNAM         |   1995 | 10291677053
 JAPAN           | VIETNAM         |   1995 | 10259020417
 CHINA           | INDONESIA       |   1996 | 11718329138
 VIETNAM         | INDONESIA       |   1996 | 11256192187
 JAPAN           | INDONESIA       |   1996 | 11196557220
 INDONESIA       | JAPAN           |   1996 | 11107350490
 CHINA           | JAPAN           |   1996 | 11034434949
 INDONESIA       | INDONESIA       |   1996 | 11031182183
 INDIA           | INDONESIA       |   1996 | 11009006499
 VIETNAM         | INDIA           |   1996 | 10947513795
 VIETNAM         | JAPAN           |   1996 | 10927373056
 INDONESIA       | CHINA           |   1996 | 10832876231
 INDIA           | INDIA           |   1996 | 10803769909
 INDIA           | CHINA           |   1996 | 10799869583
 INDONESIA       | INDIA           |   1996 | 10756005575
 CHINA           | INDIA           |   1996 | 10658108154
 VIETNAM         | CHINA           |   1996 | 10609181278
 CHINA           | VIETNAM         |   1996 | 10431194088
 CHINA           | CHINA           |   1996 | 10397959049
 INDIA           | JAPAN           |   1996 | 10367024603
 JAPAN           | CHINA           |   1996 | 10360789892
 VIETNAM         | VIETNAM         |   1996 | 10276179131
 INDIA           | VIETNAM         |   1996 | 10250670880
 JAPAN           | JAPAN           |   1996 | 10238658407
 JAPAN           | INDIA           |   1996 | 10207767222
 JAPAN           | VIETNAM         |   1996 | 10035187510
 INDONESIA       | VIETNAM         |   1996 |  9914928892
 JAPAN           | INDONESIA       |   1997 | 11482072101
 CHINA           | INDONESIA       |   1997 | 11432002557
 INDIA           | INDONESIA       |   1997 | 11418742656
 INDONESIA       | INDONESIA       |   1997 | 11266595254
 VIETNAM         | INDONESIA       |   1997 | 11107866642
 INDONESIA       | INDIA           |   1997 | 10901363844
 INDIA           | CHINA           |   1997 | 10857283915
 CHINA           | JAPAN           |   1997 | 10793325412
 JAPAN           | INDIA           |   1997 | 10773028622
 CHINA           | CHINA           |   1997 | 10772225760
 INDONESIA       | CHINA           |   1997 | 10699078012
 INDONESIA       | JAPAN           |   1997 | 10697685336
 VIETNAM         | INDIA           |   1997 | 10659789598
 INDIA           | INDIA           |   1997 | 10594944414
 CHINA           | INDIA           |   1997 | 10558080684
 INDONESIA       | VIETNAM         |   1997 | 10538588657
 VIETNAM         | JAPAN           |   1997 | 10508662011
 CHINA           | VIETNAM         |   1997 | 10419891376
 INDIA           | JAPAN           |   1997 | 10417917258
 JAPAN           | CHINA           |   1997 | 10380170603
 VIETNAM         | CHINA           |   1997 | 10347399889
 JAPAN           | JAPAN           |   1997 | 10118734544
 JAPAN           | VIETNAM         |   1997 |  9925115976
 VIETNAM         | VIETNAM         |   1997 |  9874907444
 INDIA           | VIETNAM         |   1997 |  9862710061
(150 rows)

